app/awslib.py

```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
+                                tg_arn = tg['TargetGroupArn']
+                                target_type = tg['TargetType']
+                                instances.extend(_get_instances_for_target_group(elbv2, tg_arn, target_type, region))
+    else:
+        print('Target Type is: %s - unhandled' % target_type)
+    return list(set(instances))
+
+
+# Get the public IPs for the given instances
+def _get_instances_public_ip(ec2_client, instances):
+    instance_ips = []
+    reservations = ec2_client.describe_instances(InstanceIds=instances)['Reservations']
+    for r in reservations:
+        for instance in r['Instances']:
+            if 'PublicIpAddress' in instance:
+                instance_ips.append(instance['PublicIpAddress'])
+            else:
+                print("The instance %s has no public IP" % str(instance['InstanceId']))
+    return instance_ips
+
+
 # IPs of running instances
-def _instance_ip(lb_name, region):
-    print ("Connecting to ec2 elb...")
-    elb = boto3.client('elb', region_name=region)
-    print ("Connected!")
-    print ("Retrieving load balancers...")
-    all_lbs = elb.describe_load_balancers()['LoadBalancerDescriptions']
+def list_instance_ips(lb_name, region):
+    print ("Getting IPs for instances behind load balancer %s..." % lb_name)
     instance_ips = []
-    for lb in all_lbs:
+    lb_found = False
+    print ("Connecting to ec2 elb v1...")
+    elbv1 = boto3.client('elb', region_name=region)
+    print ("Connected!")
+    print ("Retrieving classic load balancers...")
+    v1_lbs = _get_v1_lbs(elbv1, next_marker=None)
+    ec2_client = boto3.client('ec2', region_name=region)
+    for lb in v1_lbs:
         if lb_name in lb['LoadBalancerName'].lower():
-            #lb_name = str(lb['LoadBalancerName'])
+            lb_found = True
             print ("Processing instances for ELB %s" % lb['LoadBalancerName'])
             instances = [inst['InstanceId'] for inst in lb['Instances']]
             print ("Instances discovered: %s" % str(instances))
-
             if instances:
-                ec2 = boto3.client('ec2', region_name=region)
-                reservations = ec2.describe_instances(InstanceIds=instances)['Reservations']
-                for r in reservations:
-                    for instance in r['Instances']:
-                        if 'PublicIpAddress' in instance:
-                            instance_ips.append(instance['PublicIpAddress'])
-                        else:
-                            print ("The instance %s has no public IP" % str(instance['InstanceId']))
+                instance_ips.extend(_get_instances_public_ip(ec2_client, instances))
+    # Only look at v2 load balancers if we haven't already found the load balancer above
+    if not lb_found:
+        print("Didn't find the load balancer in the list of classic load balancers")
+        print ("Connecting to ec2 elb v2...")
+        elbv2 = boto3.client('elbv2', region_name=region)
```

Comment:
maybe check this as well as it could possible mess up the later calls

Reply:
Sure.